### PR TITLE
modified gradient clip norm & learning rate policy

### PR DESCRIPTION
**Pull request by yang1fan2**
Hi

My mxnet rnn modification achieved 103.7 on validation.
There are three modifications:
1. add dropout
2. Learning rate is initially set to 1.0 and halved if the perplexity does not decrease by more than 1.0 on the validation set after an epoch.
3. clip the norm of the gradients (normalized by minibatch size) at 5

Epoch [32732] Train: NLL=4.494, Perp=89.517
Epoch [32760] Train: NLL=4.496, Perp=89.678
Epoch [32788] Train: NLL=4.498, Perp=89.866
Epoch [32816] Train: NLL=4.501, Perp=90.100
Epoch [32844] Train: NLL=4.501, Perp=90.082
Epoch [32872] Train: NLL=4.503, Perp=90.284
Epoch [32900] Train: NLL=4.500, Perp=90.008
Epoch [32928] Train: NLL=4.494, Perp=89.483
Epoch [32956] Train: NLL=4.488, Perp=88.981
Epoch [32984] Train: NLL=4.490, Perp=89.098
Epoch [33012] Train: NLL=4.487, Perp=88.852
Epoch [33040] Train: NLL=4.484, Perp=88.623
Epoch [33068] Train: NLL=4.478, Perp=88.091
Epoch [33096] Train: NLL=4.477, Perp=87.950
Epoch [33124] Train: NLL=4.474, Perp=87.699
Epoch [33152] Train: NLL=4.476, Perp=87.855
Iter [24] Train: Time: 266.847 sec, NLL=4.478, Perp=88.077
Iter [24] Val: NLL=4.642, Perp=103.704
Reset learning rate to 0.00195312

Please refer to https://github.com/yoonkim/lstm-char-cnn
Thanks;

### CODE REVIEW

**Comment by jermainewang** [comment]
Hi,

Thanks for your contribution. For the gradient clipping, I think it should compute the L2 norm of all the gradients together right? In your code, you compute norm of each parameter gradient and clip them if its norm is greater than some constant, but the magnitute of different paramter will lead to different magnitute of gradient. In your lua code, you also compute the norm of all gradients together. Another thing is the `max_grad_norm`, could you also expose it as a parameter like what you did in your lua code?

Thank you,
Minjie

**Comment by yang1fan2** [comment]
@jermainewang 
Sorry for my poor expression. For clarification, it is Yoon Kim's lua code. 
Nice suggestion, I'll make these changes
Thanks;

**Comment by tqchen** [comment]
@antinucleon Can you review it and merge if you think it is good to go?

**Comment by antinucleon** [comment]
lgtm, Thanks for contribution!
